src/artm/core/batch_manager.cc

```diff
@@ -23,6 +23,23 @@ void BatchManager::DisposeModel(const ModelName& model_name) {
   in_progress_.erase(model_name);
 }
 
+void BatchManager::AddAndNext(const BatchManagerTask& task) {
+  boost::lock_guard<boost::mutex> guard(lock_);
+  tasks_.push_back(task);
```

Comment:
>tasks_.push_back(task);
This line is redundant, please remove. Note that BatchManager::Next erases the task it returns from tasks_ list.